pkg/kubelet/kubelet_node_status_test.go

```diff
@@ -1151,7 +1151,7 @@ func TestRegisterWithApiServer(t *testing.T) {
 	}
 	kubelet.machineInfo = machineInfo
 
-	done := make(chan struct{})
+	done := make(chan struct{}, 1)
```

Comment:
This makes the timeout race with the goroutine. IMO it's better to, in the timeout case, `go func() { <-done}()`.

Actually, even easier-- just make the goroutine close the channel rather than write to it.

Reply:
`close(stop)` never blocks, and unlocks an infinite number of potential future readers, and makes it obvious to a reader of the code that the channel will not be used again.

A send only unblocks one reader and says to the casual human code-reader that you intend to send another value on the channel in the future (otherwise, you would close it).

Adding a buffer to the channel makes it harder to get stuck in a weird place, but also makes it easy for real issues to go unnoticed--for the same reason. It's my least favorite solution.

Yeah, on re-review, I stand by my recommendations to use close() everywhere in this PR.